### PR TITLE
fortressVillage added

### PR DESCRIPTION
**Pull request by ujjman**
This pull request adds a village in gooey's dialog and you can spawn it easily.

### CODE REVIEW

File: assets/prefabs/structures/fortressvillage/fortressVillageBuilding/fortressVillageCattleBarn.prefab [hunk 1]
```diff
@@ -0,0 +1,90 @@
+{
+    "StructureTemplate": {
+        "type": "GooeysQuests:fortressVillageBuilding",
+        "spawnChance": 100
+    },
+    "SpawnBlockRegions": {
+        "regionsToFill": [
+     { "blockType": "CoreBlocks:Plank:engine:slope.RIGHT", "region": { "min": [9, 6, 0], "size": [1, 1, 20]}},
```
Comment (Cervator) [inline assets/prefabs/structures/fortressvillage/fortressVillageBuilding/fortressVillageCattleBarn.prefab]: Trivial: think something went wrong with indentation here (should be deeper). Might leave a bracket or two near the end out of whack too. And ideally newline at the end of every text file or GitHub shows a funny little red symbol at the end :-)

File: module.txt [hunk 2]
```diff
@@ -9,9 +9,13 @@
         {"id" : "CoreBlocks", "minVersion" : "1.0.0"},
         {"id" : "Core", "minVersion" : "2.0.0"},
         {"id" : "StructureTemplates", "minVersion" : "0.3.0"},
+        {"id" : "StructuralResources", "minVersion" : "2.0.0"},
         {"id" : "Pathfinding", "minVersion" : "0.3.0"},
+        {"id" : "engine", "minVersion" : "2.2.1"},
```
Comment (Cervator) [inline module.txt]: Don't need to declare the engine as a dependency, that's automatic 👍

File: src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    @In
+    private StructureTemplateProvider structureTemplateProvider;
+
+    private Map<EntityRef, FoundSpawnPossiblity> questToFoundSpawnPossibilityMap = new HashMap<>();
+
+    private Random random = new Random();
+
+    private Predicate<Block> isAirCondition;
+    private Predicate<Block> isGroundCondition;
+
```
Comment (Cervator) [inline src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java]: Occasional double empty lines like these don't really add anything, one blank line is enough :-)

File: src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        Vector3i randomPosition = new Vector3i(questOwnerBlockPos);
+        randomPosition.addX(randomHorizontalOffset());
+        randomPosition.addZ(randomHorizontalOffset());
+
+        Vector3i surfaceGroundBlockPosition = findSurfaceGroundBlockPosition(randomPosition);
+        if (surfaceGroundBlockPosition == null) {
+            return;
+        }
+
+        EntityRef etranceSpawner = structureTemplateProvider.
```
Comment (Cervator) [inline src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java]: Typo: `entranceSpawner` I figure?

File: src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    /**
+     * Free the memory once the quest is no longer loaded
+     */
+    @ReceiveEvent
+    public void onDeactivateQuestEntity(BeforeDeactivateComponent event, EntityRef questEntity,
+                                        PersonalQuestsComponent questsComponent) {
+        questToFoundSpawnPossibilityMap.remove(questEntity);
+    }
+
+    private Vector3i findSurfaceGroundBlockPosition(Vector3i position) {
```
Comment (Cervator) [inline src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java]: Being that this method (and maybe others) appears duplicated between quests maybe there's a way to centralize some of the code?

**Comment by ujjman** [comment]
Hey mentors I have made the changes that were told me.Also my fortress outer walls looked dull and were not very attractive so I have also made it look much impressive and attractive.

**Comment by casals** [comment]
Requested changes are OK - merging

**Pull request merged**